Products/Zuul/facades/__init__.py

```diff
@@ -208,33 +207,49 @@ def getDeviceBrains(self, uid=None, start=0, limit=50, sort='name',
             startp = 0
             limitp = None
 
-
         catbrains = cat.search(
                 'Products.ZenModel.Device.Device', start=startp,
                 limit=limitp, orderby=orderby, reverse=reverse,
                 query=query, globFilters=globFilters, hashcheck=hashcheckp)
 
-        ## Handle Production State separately
+        #
+        # Handle Production State separately
+        #
         if useProdStates:
+
+            def unbrainList(brains):
+                return [unbrain(b) for b in catbrains]
+
             psManager = IProdStateManager(self._dmd)
+
+            # This flag is used to determine if the SearchResults class needs to unbrain the provided list
+            # of entities.  If areBrains == True then it does need to unbrain the entities  This is will
+            # prevent unbraining entities twice which could be a performance issue.
+            areBrains = catbrains.areBrains
+
             # Filter by production state
             if prodStates:
-                psFilteredbrains = [brain for brain in catbrains if psManager.getProductionStateFromGUID(brain.uuid) in prodStates]
-                totalCount = len(psFilteredbrains)
+                # we need to unbrain these objects here so we have access to the getUUID() method.
+                catbrains = unbrainList(catbrains)
+                # Unbraining these brains makes them no longer brains - so set the SearchResults flag accordingly
+                areBrains = False
+
+                psFilteredBrains = [u for u in catbrains if psManager.getProductionStateFromGUID(u.getUUID()) in prodStates]
```

Comment:
since we have unbrained them, we can just do u.getProductionState() and eliminate the need for psManager.